plugins/CoreUpdater/Commands/Update.php

```diff
@@ -358,20 +358,19 @@ private function makeUpdaterInstance(OutputInterface $output)
      */
     protected function writeAlertMessageWhenCommandExecutedWithUnexpectedUser(OutputInterface $output)
     {
-        if(SettingsServer::isWindows()) {
+        if (SettingsServer::isWindows()) {
             // does not work on windows
             return;
         }
 
         $processUserAndGroup = Filechecks::getUserAndGroup();
         $fileOwnerUserAndGroup = Filechecks::getOwnerOfPiwikFiles();
 
-        if($processUserAndGroup == $fileOwnerUserAndGroup) {
+        if (!$fileOwnerUserAndGroup || $processUserAndGroup == $fileOwnerUserAndGroup) {
```

Comment:
We only compare when `getOwnerOfPiwikFiles()` returns a value. I noticed `getOwnerOfPiwikFiles()` was before already sometimes returning an empty string meaning this check might have shown the warning even though everything was fine.